### ISSUE TITLE
Move upgrade instructions to a separate page

### ISSUE THREAD

**Issue by porunov** [JanusGraph/janusgraph#1815]
Currently upgrade instructions are listed in `Changelog` but it isn't right because of the following facts:
- Upgrade instructions in most cases should upgrade from `feature` versions and not `bugfix` versions. We have currently 6 upgrade instructions: `from Titan to JanusGraph 0.1.x`, `from 0.1.x to 0.2.x`, `from 0.2.0 to 0.2.1` (exceptional case), `from 0.2.x to 0.3.x`, `from 0.3.x to 0.4.x`, `from 0.4.x to 0.5.x`.
- Currently upgrade instructions are hard to be find because you should go to `Changelog` and find the fist version of a specific feature version to which you would like to upgrade. It isn't intuitively as for me.

I would like to propose to move all upgrade instructions to a separate page.

**Comment by FlorianHockmann** [comment]
Can you explain why the upgrade instructions are hard to find in the changelog? Wouldn't users go to the changelog of the version they want to upgrade to and then get all information they need in one place there? I for example don't need information about the upgrade to version 0.3.0 if I want to update to 0.4.0,

**Comment by porunov** [comment]
> Can you explain why the upgrade instructions are hard to find in the changelog? Wouldn't users go to the changelog of the version they want to upgrade to and then get all information they need in one place there? I for example don't need information about the upgrade to version 0.3.0 if I want to update to 0.4.0

For example, I am running JanusGraph v0.1.1, I would like to upgrade to v0.3.2.
I go to changelog and see that there is no `Upgrade Instructions` for version v0.3.2. I assume, I can just increase the version in my dependencies and that's it. 
How can I know that some feature and bugfix versions had breaking changes?
Or, assuming if I ran version v0.1.1 and would like to upgrade to v0.2.3. Let's assume I knew that our `feature versions` had some breaking changes. I know that I need to go to version v0.2.0 to see upgrade instructions for version v0.2.3. I am following those instructions, everything is OK but I didn't know that there were breaking changes between v0.2.0 and v0.2.1, so my v0.2.3 doesn't work correctly because I didn't follow all upgrade instructions for all versions which were released from `0.1.1`.
Of course we can make a statement somewhere that to upgrade from a lower version to upper version you must go through all versions in changelog (including minor versions) and follow upgrade instructions for each janusgraph versions. But isn't it too many redundant information to read (i.e. release notes, tested compatibility, milestones for each version which you are not interested in)? 

I would expect more user friendly upgrade instructions. For example, the first statement for upgrade instructions for `0.3.x` version would be, `before you can upgrade to this version you should run 0.2.1 or above version` (and link to that upgrade instructions for 0.2.1). There would be upgrade instructions for `0.2.1` with the first statement `before you can upgrade to this version you should run version 0.2.0`.

Currently, I think it isn't intuitively because when the user goes to specific version, there could be no upgrade instructions (even so the user should follow some upgrade instructions).

**Comment by FlorianHockmann** [comment]
> How can I know that some feature and bugfix versions had breaking changes?

This seems to be what makes the upgrade instructions now harder to understand in your opinion if I got your point right. But I don't quite understand it as we simply don't make any breaking changes in bugfix version, only in version like `0.2.0`, `0.3.0`, and so on which is basically standard practice.
I think users have to expect from any software project that there might be breaking changes in a version like that, so they have to consult the changelog / upgrade instructions for that specific version, but not for a version like `0.2.1`.

That also simplifies the case for your example:

> assuming if I ran version v0.1.1 and would like to upgrade to v0.2.3.

as you only have to check the entry for `0.2.0` here to see if there are upgrade instructions relevant for you. `0.2.1` - `0.2.3` can't have any breaking changes which means that you can update directly to `0.2.3` and only have to consider the upgrade instructions of `0.2.0` here.

**Comment by porunov** [comment]
@FlorianHockmann you are right about bugfix upgrades. But generally our upgrade instructions isn't intuitively clear. Non of our major software dependencies (tinkerpop, scylladb, elasticsearch, hbase etc.) use this style for upgrade instructions. They all have separate `Upgrade instructions` pages where you can follow upgrade instructions without seeing changelogs.
Also, even so out page is called `Changelog` it just shows tested compatibility (matrix for `feature` versions and list for `bugfix` versions) in addition to upgrade instructions + links to the closed issues and PRs.
It is hard to call it `Changelog` because changelog doesn't include those information. Instead of that, changelog includes information about bugfix, improvments, features of specific versions.
My suggestion is to follow common style / practices that are clear for all users instead of using uncommon style for `changelog / upgrade instructions` which are used right now.
